Flickr/Flickr/GalleryViewingCollectionViewController.swift

```diff
@@ -8,12 +8,64 @@
 
 import UIKit
 
-class GalleryViewingCollectionViewController: UIViewController {
+class GalleryViewingCollectionViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
+    var photoGalleryData = [Photo]()
+    var main = ImageCollectionViewController()
+    var indexCell: IndexPath?
     
-    var photoGalleryData: [Photo]?
-
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        self.view.addSubview(collectionView)
```

Comment:
this line is hardcoded addSubview. This is already called by view controller init.